go/otgrpc/shared.go

```diff
@@ -18,6 +20,7 @@ type metadataReaderWriter struct {
 }
 
 func (w metadataReaderWriter) Set(key, val string) {
+	key = strings.ToLower(key)
```

Comment:
This isn't quite right as it violates the expectations/contract around the `TextMap` carrier format (which has case-sensitive keys).

The right thing would be to use the `HTTPHeader` format instead in which keys are case-insensitive for obvious reasons. https://github.com/bajb/grpc-opentracing/blob/7b2191ff4e3fdfd095523bc37d3da5df45b2aef9/go/otgrpc/client.go#L55

This is tricky, though, as we also need to make the parallel change to not just the reader/server side of this `metadataReaderWriter` class, but also a parallel change in other GRPC ports.

If you speak Java, it will be an easy and analogous change. For better or worse, we haven't done other languages yet, so thankfully the compatibility issues won't be that annoying.